buildprocess/svgs/SvgSpriteWebpackPlugin.js

```diff
@@ -0,0 +1,233 @@
+const SVGSpriter = require("svg-sprite");
+const collector = require("./svg-sprite-collector");
+const { sources } = require("webpack");
+const { basename, join } = require("path");
+const crypto = require("crypto");
+const fs = require("fs");
+
+/**
+ * SVG Sprite Webpack Plugin
+ * Hooks into compilation, accesses collector, compiles sprite sheet, and emits sprite.svg
+ */
+class SvgSpriteWebpackPlugin {
+  constructor(options = {}) {
+    this.options = {
+      ...options,
+      debug: options.debug || false
+    };
+
+    this.sprites = new Map();
+    this.spritesPaths = new Set();
+  }
+
+  apply(compiler) {
+    console.log("🔨 Initializing SVG Sprite Webpack Plugin...");
+    compiler.hooks.thisCompilation.tap(
+      SvgSpriteWebpackPlugin.name,
+      (compilation) => {
+        // Hook into finishModules to process after all modules are loaded
+        compilation.hooks.finishModules.tapPromise(
+          {
+            name: "SvgSpriteWebpackPlugin",
+            stage: compiler.webpack.Compilation.PROCESS_ASSETS_STAGE_ADDITIONS
+          },
+          async () => {
+            try {
+              // Check if any SVG files changed this build
+              if (!collector.hasChanges()) {
+                console.log(
+                  "⏭️  Skipping sprite compilation - no SVG changes detected"
+                );
+                return;
+              }
+              console.log("🔨 Compiling SVG sprites...");
+
+              await this.processIcons(compilation);
+
+              const script = this.generateInjectionScript();
+
+              const { sources } = compiler.webpack;
+
+              compilation.emitAsset(
+                "svg-sprite.js",
+                new sources.RawSource(script)
+              );
+
+              const mainChunk = compilation.namedChunks.get("main");
+              if (mainChunk) {
+                mainChunk.files.add("svg-sprite.js");
+              }
```

Comment:
The re-org seems to have broken this step, and so does not include `svg-sprite.js` in the final build. I think it is because this dependency add needs to happen in `compilation.hooks.processAssets` stage which is called after all chunks have been built. Here's [an example](https://github.com/orgs/webpack/discussions/18084).